modules/charset/charset.go

```diff
@@ -137,16 +138,39 @@ func DetectEncoding(content []byte) (string, error) {
 	} else {
 		detectContent = content
 	}
-	result, err := textDetector.DetectBest(detectContent)
+
+	// Now we can't use DetectBest or just results[0] because the result isn't stable - so we need a tie break
+	results, err := textDetector.DetectAll(detectContent)
 	if err != nil {
+		if err == chardet.NotDetectedError && len(setting.Repository.AnsiCharset) > 0 {
+			log.Debug("Using default AnsiCharset: %s", setting.Repository.AnsiCharset)
+			return setting.Repository.AnsiCharset, nil
+		}
 		return "", err
 	}
+
+	topConfidence := results[0].Confidence
+	topResult := results[0]
+	priority, has := setting.Repository.DetectedCharsetScore[strings.ToLower(strings.TrimSpace(topResult.Charset))]
+	for _, result := range results {
+		if result.Confidence == topConfidence {
+			resultPriority, resultHas := setting.Repository.DetectedCharsetScore[strings.ToLower(strings.TrimSpace(result.Charset))]
```

Comment:
In the future we could attempt to normalize our list casing to the lib's casing in order to avoid calling `ToLower()` in a loop.

Reply:
Hehehe have you looked at the libs casing? I copied the names exactly in to the setting, app.ini.sample and config cheat sheet. There's literally no fixed pattern for them.